src/components/Store.react.js

```diff
@@ -4,7 +4,9 @@ import PropTypes from 'prop-types';
 
 function dataCheck(data, old) {
     // Assuming data and old are of the same type.
-    if (R.isNil(old) || R.isNil(data)) {
+    const oldNull = R.isNil(old);
+    const newNull = R.isNil(data);
+    if ((oldNull || newNull) && !(oldNull && newNull)) {
```

Comment:
I found this a bit hard to read... how about `if (oldNull !== newNull)` ? Or I suppose you could short-circuit the logic below with something like `if (oldNull || newNull) { return oldNull !== newNull; }`

Separately, it looks to me as though the checks below implicitly assume `old` and `data` have the same type, and may fail if for example `data` is an `Array` and `old` is an `Object`. Should we include a step `if (R.type(old) !== type) { return false; }`?

Reply:
> Separately, it looks to me as though the checks below implicitly assume old and data

That's the second part of #422, I'll add a test and fix it in another PR. Initial implementation assumed data was of the same shape.

https://github.com/plotly/dash-core-components/blob/fa82eeb17ee7831d54fa08ea84f2054e7b5fd027/src/components/Store.react.js#L6